### ISSUE TITLE
Cannot load the domain address of the format containing ":".

### ISSUE THREAD

**Issue by Stump-Mt-Firewood** [openzim/zimit#435]
{"timestamp":"2024-11-21T07:16:22.933Z","logLevel":"warn","context":"general","message":"Invalid Page - URL must start with http:// or https://","details":{"url":"javascript:print();","page":"https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg","workerid":1}}
{"timestamp":"2024-11-21T07:16:23.029Z","logLevel":"info","context":"pageStatus","message":"Page Finished","details":{"loadState":3,"page":"https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg","workerid":1}}

command:
`docker run  -v /output:/output ghcr.io/openzim/zimit:2.1.6 zimit -u https://supermemo.guru/ --name SuperMemoguru_save_10 --waitUntil load --timeout 90 --behaviors autoplay,autofetch,siteSpecific,autoscroll --behaviorTimeout 90 --workers 4  --title SuperMemo_guru --description "SuperMemo_guru website save package" --creator Stump --source https://supermemo.guru/ --favicon https://supermemo.guru/resources/assets/SuperMemory-Guru-logo.png`

Unable to load domain addresses containing <:>, including past edit pages and web images .

e.g. 

1. https://supermemo.guru/index.php?title=Special:
2. https://supermemo.guru/wiki/Special:RecentChangesLinked?hidebots=1&target=Irreversible_maladaptability&limit=50&days=7&enhanced=1&urlversion=2"
3. https://supermemo.guru/wiki/Special:WhatLinksHere/Myth:_We_remember_useful_things_because_we_use_them"

Bug report is: 
"Invalid Page - URL must start with http:// or https://"

I tried many times, and only the command without any setting, I used when I first started using zimit was able to get some of the images, but most still didn't show up:

`docker run  -v /output:/output ghcr.io/openzim/zimit:2.1.6 zimit --url https://supermemo.guru/`

Moreover,
I tried zimit online,
It's also unretrievable, even less than the amount of data my local operation retrieves .(max = 97MB)

All of the above attempts fetched a maximum of 137.9MB (local crwal)

**Comment by benoit74** [comment]
The `Invalid Page - URL must start with http:// or https://` is just a warning, meaning than on current page a link has been found which will not be explored. I.e. in your example, `javascript:print();` URL which has been found on `https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg` will not be explored. But the page itself (https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg) is properly saved.

Regarding the fact that most images didn't show up and final ZIM size is very unpredictable / ZIM is incomplete, this is linked to another issue. Can you share the whole log? We need to look at the final message at the end of the crawl (just before starting warc2zim for the second time, the first one being at the very beginning of the log)

All that being said, the website you're trying to ZIM is a mediawiki, and we do not recommend to use zimit, see https://github.com/openzim/zimit/wiki/Frequently-Asked-Questions#can-i-use-zimit-to-zim-a-mediawiki

**Comment by Stump-Mt-Firewood** [comment]
> The `Invalid Page - URL must start with http:// or https://` is just a warning, meaning than on current page a link has been found which will not be explored. I.e. in your example, `javascript:print();` URL which has been found on `https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg` will not be explored. But the page itself (https://supermemo.guru/wiki/Special:RecentChangesLinked/File:Learning_process_simulation.jpg) is properly saved.
> 
> Regarding the fact that most images didn't show up and final ZIM size is very unpredictable / ZIM is incomplete, this is linked to another issue. Can you share the whole log? We need to look at the final message at the end of the crawl (just before starting warc2zim for the second time, the first one being at the very beginning of the log)
> 
> All that being said, the website you're trying to ZIM is a mediawiki, and we do not recommend to use zimit, see https://github.com/openzim/zimit/wiki/Frequently-Asked-Questions#can-i-use-zimit-to-zim-a-mediawiki

OK，Just now I was redoing the web archive, I got it done, successful output(138.4MB). Similar log. Here is the log information:


```
{"timestamp":"2024-11-21T10:04:24.301Z","logLevel":"info","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":6932,"total":6932,"pending":0,"failed":0,"limit":{"max":0,"hit":false},"pendingPages":[]}}
{"timestamp":"2024-11-21T10:04:24.303Z","logLevel":"info","context":"general","message":"Crawling done","details":{}}
{"timestamp":"2024-11-21T10:04:24.306Z","logLevel":"info","context":"general","message":"Exiting, Crawl status: done","details":{}}
[zimit::2024-11-21 10:04:24,338] INFO:
[zimit::2024-11-21 10:04:24,338] INFO:----------
[zimit::2024-11-21 10:04:24,338] INFO:Processing WARC files in/at /output/.tmp_snilttj/collections/crawl-20241121081438654/archive
[zimit::2024-11-21 10:04:24,338] INFO:Calling warc2zim with these args: ['--name', 'SuperMemoguru_save_11', '--scraper-suffix', 'zimit 2.1.6', '--output', '/output', '--url', 'https://supermemo.guru/', '/output/.tmp_snilttj/collections/crawl-20241121081438654/archive']
[warc2zim::2024-11-21 10:04:24,347] WARNING:HTTP 301 occurred on main page; replacing ZimPath(supermemo.guru/) with ZimPath(supermemo.guru/wiki/Main_Page)
[warc2zim::2024-11-21 10:04:30,275] INFO:Expecting 5289 ZIM entries to files
[warc2zim::2024-11-21 10:04:30,276] INFO:Expecting 5294 ZIM entries including redirects
[warc2zim::2024-11-21 10:04:33,473] WARNING:Unable to download/use favicon at https://supermemo.guru/favicon.ico
Traceback (most recent call last):
  File "/app/zimit/lib/python3.12/site-packages/warc2zim/converter.py", line 799, in _download_icon_for_url
    if not stream_file(favicon_url, byte_stream=dst)[0]:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/zimit/lib/python3.12/site-packages/zimscraperlib/download.py", line 212, in stream_file
    resp.raise_for_status()
  File "/app/zimit/lib/python3.12/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://supermemo.guru/favicon.ico
[warc2zim::2024-11-21 10:04:33,474] WARNING:No suitable illustration found, using default
Resolve redirect
set index
[zimit::2024-11-21 10:07:39,499] INFO:
[zimit::2024-11-21 10:07:39,500] INFO:----------
[zimit::2024-11-21 10:07:39,500] INFO:Cleanup, removing temp dir: /output/.tmp_snilttj

```



Command is:



```
docker run  -v /output:/output ghcr.io/openzim/zimit:2.1.6 zimit --url https://supermemo.guru/ --name SuperMemoguru_save_11 --timeout 90 --behaviorTimeout 90 --workers 8
```



I will redo the web archive with MWoffliner program, I think my question has been solved in another way, thanks for your support.

**Comment by benoit74** [comment]
OK, I'll close this issue for now, do not hesitate to ask for reopen if needed
